components/constellation/constellation.rs

```diff
@@ -1867,6 +1870,15 @@ where
             ScriptToConstellationMessage::FinishJavaScriptEvaluation(evaluation_id, result) => {
                 self.handle_finish_javascript_evaluation(evaluation_id, result)
             },
+            ScriptToConstellationMessage::WebDriverInputComplete(msg_id) => {
+                if let Some(ref reply_chan) = self.webdriver.input_command_response_channel {
+                    reply_chan
+                        .send(WebDriverCommandResponse::WebDriverInputComplete(msg_id))
+                        .unwrap_or_else(|_| {
+                            warn!("Failed to send WebDriverInputComplete {:?}", msg_id);
+                        });
+                }
```

Comment:
Theoretically at this point you can unset `self.webdriver.input_command_response_channel`, right? If you don't do this, you always leak at least one file descriptor as the sending side of the channel will never close.